Missing zap-cli fails arm64 builds in v1.1.0.1

The arm64 build of v1.1.0.1 fails with the following error:
```
:: [32/595] ACTION //third_party/connectedhomeip/src/controller/data_model:data_model_zapgen_zap_pregen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: FAILED: gen/third_party/connectedhomeip/src/controller/data_model/zap_pregen/zap-generated/access.h gen/third_party/connectedhomeip/src/controller/data_model/zap_pregen/zap-generated/gen_config.h gen/third_party/connectedhomeip/src/controller/data_model/zap_pregen/zap-generated/endpoint_config.h gen/third_party/connectedhomeip/src/controller/data_model/zap_pregen/zap-generated/CHIPClientCallbacks.h gen/third_party/connectedhomeip/src/controller/data_model/zap_pregen/zap-generated/CHIPClusters.h gen/third_party/connectedhomeip/src/controller/data_model/zap_pregen/zap-generated/IMClusterCommandHandler.cpp
:: python ../examples/chip-tool/third_party/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root ../examples/chip-tool/ --current-path ../examples/chip-tool/third_party/connectedhomeip/src/controller/data_model --default-toolchain=//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc --current-toolchain=//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc --capture-output --python-dep-list-files gen/third_party/connectedhomeip/src/controller/data_model/data_model_zapgen_zap_pregen_metadata_path_list.txt -- ../examples/chip-tool/third_party/connectedhomeip/scripts/tools/zap/generate.py --no-prettify-output --templates /build/chip-tool/parts/connectedhomeip/src/examples/chip-tool/third_party/connectedhomeip/src/app/zap-templates/app-templates.json --output-dir /build/chip-tool/parts/connectedhomeip/src/build-examples/gen/third_party/connectedhomeip/src/controller/data_model/zap_pregen/zap-generated --lock-file /build/chip-tool/parts/connectedhomeip/src/build-examples/zap_gen.lock --parallel /build/chip-tool/parts/connectedhomeip/src/examples/chip-tool/third_party/connectedhomeip/src/controller/data_model/controller-clusters.zap
:: Searching for zcl file from /build/chip-tool/parts/connectedhomeip/src/examples/chip-tool/third_party/connectedhomeip/src/controller/data_model/controller-clusters.zap
:: FAILED TO EXECUTE ZAP GENERATION: No such file or directory - "zap-cli"
:: ********************************************************************************
:: * You may need to install zap. Please ensure one of these applies:
:: * - `zap-cli` is in $PATH. Install from https://github.com/project-chip/zap/releases
:: *   see docs/guides/BUILDING.md for details
:: * - `zap-cli` is in $ZAP_INSTALL_PATH. Use this option if you
:: *   installed zap but do not want to update $PATH
:: * - Point $ZAP_DEVELOPMENT_PATH to your local copy of zap that you
:: *   develop on (to use a developer build of zap)
:: ********************************************************************************
:: [33/595] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/openthread:openthread_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [34/595] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/wpa:wpa_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [35/595] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/bluez:bluez_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [36/595] ACTION //third_party/connectedhomeip/src/platform/Linux/dbus/wpa:wpa_gen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: [37/595] ACTION //third_party/connectedhomeip/src/controller/data_model:data_model_codegen_codegen(//third_party/connectedhomeip/build/toolchain/linux:linux_arm64_gcc)
:: ninja: build stopped: subcommand failed.
```

Full build log: https://launchpadlibrarian.net/672227508/buildlog_snap_ubuntu_jammy_arm64_b17b131db72a1069b5f5c3fa87d05d81_BUILDING.txt.gz

--- a/README.md
+++ b/README.md
@@ -31,18 +31,10 @@ Discover using DNS-SD and pair:
 sudo chip-tool pairing onnetwork 110 20202021
 ```
 
-Or, pair directly by giving the IP address:
-```bash
-sudo chip-tool pairing ethernet 110 20202021 3840 192.168.1.110 5543
-```
-
 where:
 
 -   `110` is the node id being assigned to the app
 -   `20202021` is the pin code set on the app
--   `3840` is the discriminator id
--   `192.168.1.110` is the IP address of the device running the app
--   `5540` the port for server that runs inside the app
 
 ### Commissioning into Thread network over BLE
 Obtain Thread network credential:

--- a/README.md
+++ b/README.md
@@ -31,10 +31,18 @@ Discover using DNS-SD and pair:
 sudo chip-tool pairing onnetwork 110 20202021
 ```
 
+Or, pair directly by giving the IP address:
+```bash
+sudo chip-tool pairing ethernet 110 20202021 3840 192.168.1.110 5543
+```
+
 where:
 
 -   `110` is the node id being assigned to the app
 -   `20202021` is the pin code set on the app
+-   `3840` is the discriminator id
+-   `192.168.1.110` is the IP address of the device running the app
+-   `5540` the port for server that runs inside the app
 
 ### Commissioning into Thread network over BLE
 Obtain Thread network credential:

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -21,7 +21,7 @@ parts:
   connectedhomeip:
     plugin: nil
     build-environment:
-      - TAG: v1.1.0.1
+      - TAG: v1.0.0.2
     override-pull: |
       # shallow clone the project its submodules
       git clone https://github.com/project-chip/connectedhomeip.git --depth=1 --branch=$TAG .

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -21,7 +21,7 @@ parts:
   connectedhomeip:
     plugin: nil
     build-environment:
-      - TAG: v1.0.0.2
+      - TAG: v1.1.0.1
     override-pull: |
       # shallow clone the project its submodules
       git clone https://github.com/project-chip/connectedhomeip.git --depth=1 --branch=$TAG .

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -117,7 +117,6 @@ parts:
       - libgif-dev 
       - librsvg2-dev
       - wget
-      - curl
 
 apps:
   chip-tool:

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -21,7 +21,7 @@ parts:
   connectedhomeip:
     plugin: nil
     build-environment:
-      - TAG: v1.0.0.2
+      - TAG: v1.1.0.1
     override-pull: |
       # shallow clone the project its submodules
       git clone https://github.com/project-chip/connectedhomeip.git --depth=1 --branch=$TAG .
@@ -36,6 +36,37 @@ parts:
     override-build: |
       mkdir -p $CRAFT_PART_INSTALL/bin
 
+      # Install NodeJS and ZAP tool for arm builds
+      if [[ $SNAP_ARCH == "arm64" ]]; then
+        set -x
+
+        mkdir node_js
+        cd node_js
+        wget https://nodejs.org/dist/v12.22.12/node-v12.22.12-linux-x64.tar.xz
+        tar xfvJ node-v12.22.12-linux-x64.tar.xz
+        cp -rn node-v12.22.12-linux-x64/. /opt/node-v12.22.12-linux-x64/
+        rm -r node-v12.22.12-linux-x64
+        rm -rf /opt/node
+        rm -rf /usr/bin/node
+        rm -rf /usr/bin/npm
+        rm -rf /usr/bin/npx
+        ln -s /opt/node-v12.22.12-linux-x64 /opt/node
+        ln -s /opt/node/bin/* /usr/bin
+        cd ..
+        rm -rf node_js
+
+        ZAP_VERSION=v2023.05.22-nightly
+        mkdir -p /opt/zap-${ZAP_VERSION}
+        git clone https://github.com/project-chip/zap.git /opt/zap-${ZAP_VERSION}
+        cd /opt/zap-${ZAP_VERSION}
+        git checkout -b ${ZAP_VERSION}
+        npm cache clean --force
+        npm install -g npm@latest
+        npm ci
+        export ZAP_DEVELOPMENT_PATH=/opt/zap-${ZAP_VERSION}
+      fi
+
+      cd $CRAFT_PART_BUILD
       cd ../../connectedhomeip/src
       
       # The project writes its data to /tmp which isn't persisted.
@@ -80,6 +111,13 @@ parts:
       - libcairo2-dev
       - libreadline-dev
       - generate-ninja
+      - build-essential 
+      - libpango1.0-dev 
+      - libjpeg-dev 
+      - libgif-dev 
+      - librsvg2-dev
+      - wget
+      - curl
 
 apps:
   chip-tool:

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -21,7 +21,7 @@ parts:
   connectedhomeip:
     plugin: nil
     build-environment:
-      - TAG: v1.1.0.1
+      - TAG: v1.0.0.2
     override-pull: |
       # shallow clone the project its submodules
       git clone https://github.com/project-chip/connectedhomeip.git --depth=1 --branch=$TAG .
@@ -71,37 +71,6 @@ parts:
       test -f $ZAP_ENV && source $ZAP_ENV
       mkdir -p $CRAFT_PART_INSTALL/bin
 
-      # Install NodeJS and ZAP tool for arm builds
-      if [[ $SNAP_ARCH == "arm64" ]]; then
-        set -x
-
-        mkdir node_js
-        cd node_js
-        wget https://nodejs.org/dist/v12.22.12/node-v12.22.12-linux-x64.tar.xz
-        tar xfvJ node-v12.22.12-linux-x64.tar.xz
-        cp -rn node-v12.22.12-linux-x64/. /opt/node-v12.22.12-linux-x64/
-        rm -r node-v12.22.12-linux-x64
-        rm -rf /opt/node
-        rm -rf /usr/bin/node
-        rm -rf /usr/bin/npm
-        rm -rf /usr/bin/npx
-        ln -s /opt/node-v12.22.12-linux-x64 /opt/node
-        ln -s /opt/node/bin/* /usr/bin
-        cd ..
-        rm -rf node_js
-
-        ZAP_VERSION=v2023.05.22-nightly
-        mkdir -p /opt/zap-${ZAP_VERSION}
-        git clone https://github.com/project-chip/zap.git /opt/zap-${ZAP_VERSION}
-        cd /opt/zap-${ZAP_VERSION}
-        git checkout -b ${ZAP_VERSION}
-        npm cache clean --force
-        npm install -g npm@latest
-        npm ci
-        export ZAP_DEVELOPMENT_PATH=/opt/zap-${ZAP_VERSION}
-      fi
-
-      cd $CRAFT_PART_BUILD
       cd ../../connectedhomeip/src
       
       # The project writes its data to /tmp which isn't persisted.
@@ -146,12 +115,6 @@ parts:
       - libcairo2-dev
       - libreadline-dev
       - generate-ninja
-      - build-essential 
-      - libpango1.0-dev 
-      - libjpeg-dev 
-      - libgif-dev 
-      - librsvg2-dev
-      - wget
 
 apps:
   chip-tool:

